demos/test_vlc_net/src/server.rs

```diff
@@ -0,0 +1,845 @@
+use anyhow::anyhow;
+use base58::FromBase58;
+use base64::{engine::general_purpose, Engine as _};
+use bincode::Options;
+use gossip::network::{Discover, Event, Network, Sha256Topic};
+use libp2p::{
+    futures::StreamExt,
+    identity::{self, Keypair, PublicKey},
+    multiaddr, PeerId,
+};
+use serde::{Deserialize, Serialize};
+use sha2::{Digest, Sha256};
+use std::io::Write;
+use std::{
+    collections::{BTreeMap, VecDeque},
+    sync::Arc,
+};
+use std::{fs::File, str::FromStr};
+use tokio::{
+    sync::{mpsc, RwLock, Semaphore},
+    time::{self, sleep, timeout, Duration},
+};
+use tracing::{debug, error, info, warn};
+use vlc::ordinary_clock::OrdinaryClock;
+
+const _DEFAULT_PAYLOAD: &str = "Hello Hetu";
+const NET_TIMEOUT: u64 = 1; // second
+
+// Metric visualization can utilities prometheus, and grafana toolboxs.
+#[derive(Debug, Clone)]
+struct OuterMsg {
+    data: Vec<u8>,
+    topic: Sha256Topic,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct P2PMsg {
+    clock: OrdinaryClock,
+    data: String,
```

Comment:
`data` should be `Vec<u8>` not `String`

Reply:
This will adopted and updated, this is a field reserved for future expansion and is not used currently.